templates/pages/reset-password.jsx

```diff
@@ -32,28 +33,40 @@ var ResetPassword = React.createClass({
     linkQuery.state = this.state.queryObj.state;
     linkQuery.scopes = this.state.queryObj.scopes;
     linkQuery.response_type = this.state.queryObj.response_type;
+
+    var content = (
```

Comment:
this is quite a lot of allocation to throw away if `resetSuccess` is true. Does it make sense to test for that first?